File: src/test/java/calculator/Numbers.java
package calculator;

import java.util.Arrays;
import java.util.List;

public record Numbers(List<NonNegativeNumber> numbers) {

	public static Numbers from(final String[] inputCharacters) {
		return new Numbers(Arrays.stream(inputCharacters)
			.map(NonNegativeNumber::from)
			.toList());
	}

	public int sum() {
		return numbers.stream().mapToInt(NonNegativeNumber::value).sum();
	}
}

File: src/test/java/calculator/Numbers.kt
package calculator

class Numbers(
    private val numbers: List<Number>
) {
    companion object {

        fun create(splitStringNumbers: List<String>): Numbers {
            return Numbers(
                numbers = splitStringNumbers.map {
                    Number.create(it)
                }
            )
        }
    }

    fun sum(): Int {
        return numbers.sumOf { it.number }
    }
}
